app/lanes.go

```diff
@@ -0,0 +1,118 @@
+	// a transaction. Each lane can have a different signer extractor if needed.
+	signerAdapter := signerextraction.NewDefaultAdapter()
+
+	// 2. Create the configurations for each lane. These configurations determine how many
+	// transactions the lane can store, the maximum block space the lane can consume, and
+	// the signer extractor used to extract the expected signers from a transaction.
+
+	// Create a mev configuration that accepts maxTxPerMEVLane transactions and consumes 20% of the
+	// block space.
+	mevConfig := base.LaneConfig{
+		Logger:          app.Logger(),
+		TxEncoder:       txConfig.TxEncoder(),
+		TxDecoder:       txConfig.TxDecoder(),
+		MaxBlockSpace:   mevLaneBlockspacePercentage,
+		SignerExtractor: signerAdapter,
+		MaxTxs:          maxTxPerMEVLane,
+	}
+
+	// Create a free configuration that accepts maxTxPerFreeLane transactions and consumes 5% of the
+	// block space.
+	freeConfig := base.LaneConfig{
+		Logger:          app.Logger(),
+		TxEncoder:       txConfig.TxEncoder(),
+		TxDecoder:       txConfig.TxDecoder(),
+		MaxBlockSpace:   freeLaneBlockspacePercentage,
+		SignerExtractor: signerAdapter,
+		MaxTxs:          maxTxPerFreeLane,
+	}
+
+	// Create a default configuration that accepts maxTxPerDefaultLane transactions and consumes 60% of the
+	// block space.
+	defaultConfig := base.LaneConfig{
+		Logger:          app.Logger(),
+		TxEncoder:       txConfig.TxEncoder(),
+		TxDecoder:       txConfig.TxDecoder(),
+		MaxBlockSpace:   defaultLaneBlockspacePercentage,
+		SignerExtractor: signerAdapter,
+		MaxTxs:          maxTxPerDefaultLane,
+	}
+
+	// 3. Create the match handlers for each lane. These match handlers determine whether or not
+	// a transaction belongs in the lane.
+
+	// Create the final match handler for the mev lane.
+	factory := mevlane.NewDefaultAuctionFactory(txConfig.TxDecoder(), signerAdapter)
+	mevMatchHandler := factory.MatchHandler()
+
+	// Create the final match handler for the free lane.
+	freeMatchHandler := WithdrawStakingRewardsMatchHandler()
+
+	// Create the final match handler for the default lane. I.e this will direct all txs that are
+	// not free nor mev to this lane
+	defaultMatchHandler := base.DefaultMatchHandler()
+
+	// 4. Create the lanes.
+	mevLane := mevlane.NewMEVLane(
+		mevConfig,
+		factory,
+		mevMatchHandler,
+	)
+
+	freeLane := freelane.NewFreeLane(
+		freeConfig,
+		base.DefaultTxPriority(),
+		freeMatchHandler,
+	)
+
+	defaultLane := defaultlane.NewDefaultLane(
+		defaultConfig,
+		defaultMatchHandler,
+	)
+
+	return mevLane, freeLane, defaultLane
+}
+
+// WithdrawStakingRewardsMatchHandler is a match handler that matches transactions that contain a withdraw staking rewards message
+func WithdrawStakingRewardsMatchHandler() base.MatchHandler {
+	return func(ctx sdk.Context, tx sdk.Tx) bool {
+		for _, msg := range tx.GetMsgs() {
+			if _, ok := msg.(*distrtypes.MsgWithdrawDelegatorReward); ok {
+				return true
+			}
+		}
+
+		return false
+	}
+}
```

Comment:
We proposed this under direction we received from a skip x osmosis weekly chat, happy to change this further

Reply:
We probably need to ensure the amount being withdrawn hits a certain threshold to prevent this being taken advantage of right?

Reply:
Previously discussed this with @nivasan1 async.

From @nivasan1 
> So, looking into the MsgWithdrawDelegatorReward filter, we'd need to introspect state to determine what the rewards to be withdrawn are (this is not a parameter of the message), this would make this filter break key assumptions of the MatchHandler in that it's execution could differ between ingress and insertion into a Proposal. Would this option still be interesting? This may be possible if we were to Nop ProcessProposal

My reply:
> Hmm that makes sense. In the end, we would still like to move towards the non-Nop ProcessProposal. So if we were to implement this, it would have to be redesigned.
In that case, I think it is fine to leave the withdrawal as is. That is, without the conditional logic based on reward value and simply go to free lane.
I think that the original proposal for making MsgWithdawDelegatorReward in such a way was to prevent spam. However, spam is still capped at the lane size. In addition, a spammer first needs to delegate with a non-zero tx fee to be able to spam withdrawals that are limited by the lane size.

TLDR; we should be fine leaving this as is